### PR TITLE
Wrap line numbers in spans

### PR DESCRIPTION
**Pull request by twe4ked**
I also added a commit to exclude a trailing blank newline from the line numbers.

### CODE REVIEW

File: src/highlight.js [hunk 1]
```diff
@@ -43,11 +43,11 @@ function createElement({ node, style, useInlineStyles, key }) {
 }
 
 function getLineNumberString(lines, startingLineNumber) {
```
Comment (conorhastings) [inline src/highlight.js]: maybe we should change this to just getLineNumbers now?

File: src/highlight.js [hunk 2]
```diff
@@ -43,11 +43,11 @@ function createElement({ node, style, useInlineStyles, key }) {
...
-  return lines.reduce((lineCountString, _, i) => lineCountString + `${i + startingLineNumber}\n`, '');
+  return lines.map((_line, i) => { return <span className="line" key={"line-" + i}>{`${i + startingLineNumber}\n`}</span> })
```
Comment (conorhastings) [inline src/highlight.js]: can we use single quotes on the className and a template string for key 
```js
key={`line-${i}`}
```

Comment (conorhastings) [inline src/highlight.js]: The first argument here can just be a `_` as well

Comment (conorhastings) [inline src/highlight.js]: And for the className can we use something more unique, maybe scope it to the library? 
`className='react-syntax-highlighter-line-number`

File: src/highlight.js [hunk 3]
```diff
@@ -43,11 +43,11 @@ function createElement({ node, style, useInlineStyles, key }) {
...
 }
 
 function LineNumbers({ codeString, style = {float: 'left', paddingRight: '10px'}, startingLineNumber }) {
-  return <code style={style}>{getLineNumberString(codeString.split('\n'), startingLineNumber)}</code>
+  return <code style={style}>{getLineNumberString(codeString.replace(/\n$/, '').split('\n'), startingLineNumber)}</code>
```
Comment (conorhastings) [inline src/highlight.js]: What is the replace for?

File: src/highlight.js (hunk 1 above)
Comment (twe4ked) [inline src/highlight.js]: Ah, missed this. Good catch.

File: src/highlight.js (hunk 3 above)
Comment (twe4ked) [inline src/highlight.js]: So we don't get a line number for a trailing newline. See: 05df3b8f5d679bc11f95b4000c9f6f8fcb4f73c0

Comment (conorhastings) [inline src/highlight.js]: cool

File: src/highlight.js [hunk 4]
```diff
@@ -42,12 +42,12 @@ function createElement({ node, style, useInlineStyles, key }) {
   }
 }
 
-function getLineNumberString(lines, startingLineNumber) {
-  return lines.reduce((lineCountString, _, i) => lineCountString + `${i + startingLineNumber}\n`, '');
+function getLineNumbers(lines, startingLineNumber) {
+  return lines.map((_, i) => { return <span className="react-syntax-highlighter-line-number" key={`line-${i}`}>{`${i + startingLineNumber}\n`}</span> })
```
Comment (conorhastings) [inline src/highlight.js]: we can use implicit return here instead of `{` and explcit `return` statement

**Comment by conorhastings** [comment]
@twe4ked that last comment + update the tests and it looks good! thanks!

**Pull request closed without merging**